perllib/FixMyStreet/App/Controller/Auth.pm

```diff
@@ -182,12 +184,35 @@ sub token : Path('/M') : Args(1) {
 
     # find or create the user related to the token.
     my $data = $token_obj->data;
-    my $user = $c->model('DB::User')->find_or_create( { email => $data->{email} } );
+
+    my $user = $c->model('DB::User')->find_or_new({ email => $data->{email} });
+
+    if ($data->{old_email}) {
+        # Were logged in as old_email, want to switch to email ($user)
+        if ($user->in_storage) {
+            my $old_user = $c->model('DB::User')->find({ email => $data->{old_email} });
+            if ($old_user) {
+                foreach (qw(Problem Comment Alert)) {
+                    $c->model("DB::$_")
+                        ->search({ user_id => $user->id })
+                        ->update({ user_id => $old_user->id });
+                }
+                $user->delete;
```

Comment:
There's a very unlikely crash here, if the user being deleted has some `admin_log` entries.